be/src/agent/task_worker_pool.cpp

```diff
@@ -826,46 +714,41 @@ void* TaskWorkerPool::_publish_version_worker_thread_callback(void* arg_this) {
                 break;
             } else {
                 OLAP_LOG_WARNING("publish version error, retry. "
-                                 "[transaction_id=%ld, error_tablet_size=%d]",
+                                 "[transaction_id=%ld, error_tablets_size=%d]",
                                  publish_version_req.transaction_id, error_tablet_ids.size());
-                retry_time += 1;
-                sleep(1);
+                ++retry_time;
+                SleepForMs(1000);
             }
         }
 
         TFinishTaskRequest finish_task_request;
         if (res != OLAP_SUCCESS) {
-            // if publish failed, return failed, fe will ignore this error and 
-            // check error tablet ids and fe will also republish this task
-            status_code = TStatusCode::RUNTIME_ERROR;
+            DorisMetrics::publish_task_failed_total.increment(1);
+            // if publish failed, return failed, FE will ignore this error and
+            // check error tablet ids and FE will also republish this task
             LOG(WARNING) << "publish version failed. signature:" << agent_task_req.signature;
```

Comment:
better to print `res` here, and also put the `res` in the following `Status::RuntimeError()`

Reply:
Done.